core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -337,9 +337,12 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
    * This callback is invoked by the controller's LogDirEventNotificationListener with the list of broker ids who
    * have experienced new log directory failures. In response the controller should send LeaderAndIsrRequest
    * to all these brokers to query the state of their replicas
+   * to all these brokers to query the state of their replicas. Replicas with an offline log directory respond with
+   * KAFKA_STORAGE_ERROR.
```

Comment:
Perhaps we can change it slightly to "Replicas with an offline log directory respond with KAFKA_STORAGE_ERROR, which will be handled by the LeaderAndIsrResponseReceived event"?